pcsx2/GS/GS.cpp

```diff
@@ -0,0 +1,2345 @@
+{
+	GSRendererType renderer = GSRendererType::Default;
+
+	// Legacy GUI expects to acquire vsync from the configuration files.
+
+	s_vsync = theApp.GetConfigI("vsync");
+
+	if (mt == 2)
+	{
+		// pcsx2 sent a switch renderer request
+		mt = 1;
+	}
+	else
+	{
+		// normal init
+
+		renderer = static_cast<GSRendererType>(theApp.GetConfigI("Renderer"));
+	}
+
+	*dsp = NULL;
+
+	int retval = _GSopen(dsp, title, renderer);
+
+	if (retval == 0 && s_gs)
+	{
+		s_gs->SetMultithreaded(!!mt);
+	}
+
+	gsopen_done = true;
+
+	return retval;
+}
+
+void GSreset()
+{
+	try
+	{
+		s_gs->Reset();
+	}
+	catch (GSRecoverableError)
+	{
+	}
+}
+
+void GSgifSoftReset(uint32 mask)
+{
+	try
+	{
+		s_gs->SoftReset(mask);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+}
+
+void GSwriteCSR(uint32 csr)
+{
+	try
+	{
+		s_gs->WriteCSR(csr);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+}
+
+void GSinitReadFIFO(uint8* mem)
+{
+	GL_PERF("Init Read FIFO1");
+	try
+	{
+		s_gs->InitReadFIFO(mem, 1);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+	catch (const std::bad_alloc&)
+	{
+		fprintf(stderr, "GS: Memory allocation error\n");
+	}
+}
+
+void GSreadFIFO(uint8* mem)
+{
+	try
+	{
+		s_gs->ReadFIFO(mem, 1);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+	catch (const std::bad_alloc&)
+	{
+		fprintf(stderr, "GS: Memory allocation error\n");
+	}
+}
+
+void GSinitReadFIFO2(uint8* mem, uint32 size)
+{
+	GL_PERF("Init Read FIFO2");
+	try
+	{
+		s_gs->InitReadFIFO(mem, size);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+	catch (const std::bad_alloc&)
+	{
+		fprintf(stderr, "GS: Memory allocation error\n");
+	}
+}
+
+void GSreadFIFO2(uint8* mem, uint32 size)
+{
+	try
+	{
+		s_gs->ReadFIFO(mem, size);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+	catch (const std::bad_alloc&)
+	{
+		fprintf(stderr, "GS: Memory allocation error\n");
+	}
+}
+
+void GSgifTransfer(const uint8* mem, uint32 size)
+{
+	try
+	{
+		s_gs->Transfer<3>(mem, size);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+}
+
+void GSgifTransfer1(uint8* mem, uint32 addr)
+{
+	try
+	{
+		s_gs->Transfer<0>(const_cast<uint8*>(mem) + addr, (0x4000 - addr) / 16);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+}
+
+void GSgifTransfer2(uint8* mem, uint32 size)
+{
+	try
+	{
+		s_gs->Transfer<1>(const_cast<uint8*>(mem), size);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+}
+
+void GSgifTransfer3(uint8* mem, uint32 size)
+{
+	try
+	{
+		s_gs->Transfer<2>(const_cast<uint8*>(mem), size);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+}
+
+void GSvsync(int field)
+{
+	try
+	{
+#ifdef _WIN32
+
+		if (s_gs->m_wnd->IsManaged())
+		{
+			MSG msg;
+
+			memset(&msg, 0, sizeof(msg));
+
+			while (msg.message != WM_QUIT && PeekMessage(&msg, NULL, 0, 0, PM_REMOVE))
+			{
+				TranslateMessage(&msg);
+				DispatchMessage(&msg);
+			}
+		}
+
+#endif
+
+		s_gs->VSync(field);
+	}
+	catch (GSRecoverableError)
+	{
+	}
+	catch (const std::bad_alloc&)
+	{
+		fprintf(stderr, "GS: Memory allocation error\n");
+	}
+}
+
+uint32 GSmakeSnapshot(char* path)
```

Comment:
make it an std::string or a const char*

Reply:
I am NOT about to rewrite GS in this PR, we have already way more than enough shit to go through